app/scripts/background.js

```diff
@@ -533,13 +533,8 @@ function setupController(initState, initLangCode) {
         ),
       );
 
-    // We're specifcally avoid using approvalController directly for better
-    // Error support during rejection
-    Object.keys(
-      controller.permissionsController.approvals.state.pendingApprovals,
-    ).forEach((approvalId) =>
-      controller.permissionsController.rejectPermissionsRequest(approvalId),
-    );
+    // Finally, reject all approvals managed by the ApprovalController
+    controller.approvalController.clear();
```

Comment:
I had recommended rejecting these via the `permissionsController` in this comment: https://github.com/MetaMask/metamask-extension/pull/12643#discussion_r747590236

I hadn't realized there was a `clear` method that did reject them all properly. Thought the error used by `clear` might not be especially appropriate (`ethErrors.rpc.resourceUnavailable('The request was rejected; please try again.',);`). That makes it sound like it was rejected by the system for because we had insufficient resources to handle it, rather than because the user explicitly rejected it.

Reply:
That makes sense, although with the new `PermissionController`, pending approvals are only stored in the `ApprovalController`. That said, we can improve upon this very soon: https://github.com/MetaMask/controllers/pull/656

Reply:
What do you mean by "pending approvals are only stored in the `ApprovalController`"? That was the case with the previous permission controller as well.

Reply:
Ah, I think pending permissions requests were/are also stored in `rpc-cap` under a state key in there, but I see now that we were just getting the IDs from the approval controller regardless. So, never mind that comment!

For present purposes, although I believe that the approval controller is only used for permission requests at the moment, I think it's bad to maintain code based on that assumption, and we should just use `clear` and wait for the improved error message to land.

Reply:
Sounds good!